bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -132,7 +132,7 @@ public Integer getRank() {
             @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
         for (DynamicStateDescriptionProvider provider : dynamicStateDescriptionProviders) {
             StateDescription stateDescription = provider.getStateDescription(channel, originalStateDescription, locale);
-            if (stateDescription != null) {
+            if (stateDescription != null && !stateDescription.equals(originalStateDescription)) {
```

Comment:
Question is if we should just ignore these wrong implementations in bindings or if we should throw an exception here and force the developer to fix it.

Reply:
Being completely silent about it doesn't feel right. Would the exception be thrown every time the state options are accessed? If so, that could result in a lot of exceptions. Maybe just log it instead?

Reply:
I added a ERROR log message.

Reply:
Can you also update the JavaDocs of these providers for the fact that if the result is not null it should not be the original object?

Reply:
I already did (see e.g. [here](https://github.com/openhab/openhab-core/pull/1399/files#diff-af4592feefd1fdeb2cb4cfcc431d75c0R38)).

Reply:
Great! Also nice to see some unit tests for this. :+1: